service/history/mutableStateTaskGenerator.go

```diff
@@ -204,9 +204,10 @@ func (r *mutableStateTaskGeneratorImpl) generateDelayedWorkflowTasks(
 	switch startAttr.GetInitiator() {
 	case enumspb.CONTINUE_AS_NEW_INITIATOR_RETRY:
 		workflowBackoffType = enumsspb.WORKFLOW_BACKOFF_TYPE_RETRY
-	case enumspb.CONTINUE_AS_NEW_INITIATOR_CRON_SCHEDULE,
-		enumspb.CONTINUE_AS_NEW_INITIATOR_WORKFLOW:
+	case enumspb.CONTINUE_AS_NEW_INITIATOR_CRON_SCHEDULE, enumspb.CONTINUE_AS_NEW_INITIATOR_WORKFLOW:
 		workflowBackoffType = enumsspb.WORKFLOW_BACKOFF_TYPE_CRON
+	case enumspb.CONTINUE_AS_NEW_INITIATOR_UNSPECIFIED:
```

Comment:
I think this should never happen and should return an error also. Having `WorkflowBackoffTimerTask` with `UNSPECIFIED` type looks weird.

Reply:
you are right. there was bug where this was not getting set to the right value for cron workflows. fixed